fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalJoinOperator.java

```diff
@@ -109,6 +113,20 @@ public ColumnRefSet getRequiredChildInputColumns() {
         return result;
     }
 
+    @Override
+    public ColumnRefOperator getSmallestColumns(ColumnRefFactory columnRefFactory, OptExpression expr) {
```

Comment:
I get your bug, but may we fix LogicalJoinOperator.getOutputColumns is better?

Reply:
That's more clear.

But this way would cause a lot of failed test cases, I guess changing the `getOutputColumns` semantic is dangerous.